src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -56,32 +54,14 @@ internal sealed class BuildUpToDateCheck : OnceInitializedOnceDisposedUnderLockA
         private readonly ITelemetryService _telemetryService;
         private readonly IFileSystem _fileSystem;
 
+        private State _state = State.CreateEmpty();
+
         private IDisposable? _link;
-        private IComparable? _lastVersionSeen;
 
-        private bool _isDisabled = true;
         private bool _itemsChangedSinceLastCheck = true;
-        private string? _msBuildProjectFullPath;
-        private string? _msBuildProjectDirectory;
-        private string? _markerFile;
-        private string? _outputRelativeOrFullPath;
-        private string? _newestImportInput;
 
         internal DateTime LastCheckTimeUtc { get; private set; } = DateTime.MinValue;
```

Comment:
This is the other remaining fields that is mutated by `IsUpToDateAsync`. I am not sure it's needed and may try to remove it in future.